examples/tv-casting-app/darwin/TvCasting/TvCasting/MCApplicationBasicReadVendorIDExampleViewModel.swift

```diff
@@ -30,66 +30,80 @@ class MCApplicationBasicReadVendorIDExampleViewModel: ObservableObject {
 
     func read(castingPlayer: MCCastingPlayer)
     {
+        self.Log.info("MCApplicationBasicReadVendorIDExampleViewModel.read()")
+        castingPlayer.logAllEndpoints()
+        var selectedEndpoint: MCEndpoint?
+
         // select the MCEndpoint on the MCCastingPlayer to invoke the command on
-        if let endpoint: MCEndpoint = castingPlayer.endpoints().filter({ $0.vendorId().intValue == sampleEndpointVid}).first
+        if let endpoint = castingPlayer.endpoints().filter({ $0.vendorId().intValue == sampleEndpointVid }).first {
+            selectedEndpoint = endpoint
+        // For the example Commissioner-Generated passcode commissioning flow, run demo interactions with the Endpoint with
+        // ID 1. For this flow, we commissioned with the Target Content Application with Vendor ID 1111. Since this target
+        // content application does not report its Endpoint's Vendor IDs, we find the desired endpoint based on the Endpoint
+        // ID. See connectedhomeip/examples/tv-app/tv-common/include/AppTv.h.
+        } else if let endpoint = castingPlayer.endpoints().filter({ $0.identifier().intValue == 1 }).first {
+            self.Log.info("MCApplicationBasicReadVendorIDExampleViewModel.read() No endpoint matching the sampleEndpointVid: \(String(describing: self.sampleEndpointVid)), but found endpoint with identifier: 1")
```

Comment:
Like Android/Linux, we should pass in a flag to this ViewModel: useCommissionerGeneratedPasscode = true/false, to help us run the right endpoint selection logic.

Reply:
There is an advantage in falling back since the sample vendorID 65521 is actually available here and is our preferred content app vID for sample interactions. Something might have recently changed on the TV-app side recently; I'll file a JIRA to investigate and possibly change Android/Linux to use 65521 for demo interactions as well. We also avoid passing that flag useCommissionerGeneratedPasscode, which can be confusing since after commissioning it should not matter how we commissioned the Casting Player. 

MCApplicationBasicReadVendorIDExampleViewModel.read()
MCCastingPlayer logAllEndpoints():
MCCastingPlayer MCEndpoint details: <MCEndpoint: identifier=1, vendorId=0, productId=0>
MCCastingPlayer MCEndpoint details: <MCEndpoint: identifier=2, vendorId=0, productId=0>
MCCastingPlayer MCEndpoint details: <MCEndpoint: identifier=4, vendorId=65521, productId=32768>
MCApplicationBasicReadVendorIDExampleViewModel.read() selected endpoint: <MCEndpoint: identifier=4, vendorId=65521, productId=32768>

Reply:
Adding MCEndpointSelector.swift